meeg_tools/utils/log.py

```diff
@@ -49,6 +49,7 @@ def update_log(log_file_path: str, epochs: Epochs, notes: str) -> pd.DataFrame:
             "t_min": [epochs.tmin],
             "t_max": [epochs.tmax],
             "n_interpolated": [np.NaN],
+            "excluded_components": ica.exclude,
```

Comment:
For reproducibility, saving the ICA into a file is a better way than storing the indices of the components. If you run ICA multiple times, you can expect that the order of components may not be the same -- therefore saving just the indices is not sufficient to be able to reproduce the algorithm output. 

What you can do instead:
1) In the preprocessing notebook create a new cell after applying the ICA solution, to save it:
there can be a common file path, for instance:
`preprocessed/epochs_asrt/ica/file_name-ica.fiff`

https://mne.tools/stable/generated/mne.preprocessing.ICA.html#mne.preprocessing.ICA.save

2) Incorporate saving ICA into the `apply_ica()` function:
https://github.com/weiglszonja/meeg-tools/blob/0e302337e1945c4762c5742a7e2364903df70b90/meeg_tools/preprocessing.py#L103

Reply:
1. In the preprocessing notebook, do this: 

```
# Save ICA cleaned epochs 
epochs_ica.save(os.path.join(epochs_path, f'{epochs_ica.info["temp"]}-epo.fif.gz'),
                overwrite=True)

# Save ICA solution to file
# File path will be .../.../preprocessed/epochs_asrt/ica
ica_path = os.path.join(epochs_path, "ica")
if not os.path.exists(ica_path):
    os.makedirs(ica_path)

ica.save(os.path.join(ica_path, f'{epochs_ica.info["temp"]}-ica.fif.gz'))
```

Reply:
OK, thanks @weiglszonja!